lib/public.js

```diff
@@ -36,11 +37,34 @@ class Public {
      */
     publicRequest({
         resource,
-        params
+        params,
+        post=false,
+        headers={}
     } = {}) {
+
         params.data.token = this.token;
         params.test = params.test ? 1 : 0;
         KlaviyoRequest.clean(params);
+
+        if (post) {
+            const method = 'POST';
+            const postPath = `/${API_ROOT}/${resource}`;
+            const postHeaders = headers
+            postHeaders['User-Agent'] = `Klaviyo-Node.js/${PACKAGE_VERSION}`;
```

Comment:
I believe this header should already be set downstream in the `_request` function:

https://github.com/klaviyo/node-klaviyo/blob/master/lib/request.js#L32-L40